### ISSUE TITLE
[Feature request] Add a hook to allow user to manipulate model init kwargs

### ISSUE THREAD

**Issue by nateraw** [huggingface/huggingface_hub#149]
The current `from_pretrained` implementation in `ModelHubMixin` reads in `config.json` (if available), and adds it as a kwarg named `'config'` in the model init kwargs. This is in line with `transformers` style, more or less.

I'd rather explicitly write config kwargs in my model's `__init__`. To do that, we have to unpack the config dict instead of having it be a kwarg. The solution right now is to basically copy paste 99% of `ModelHubMixin.from_pretrained` to update one line of code.

---

What do you think about adding a simple hook that wraps this line and allows users to manipulate `model_kwargs` if they'd like by overriding?

https://github.com/huggingface/huggingface_hub/blob/379414dade73e0816a996d83ead7e4ffd6986a53/src/huggingface_hub/hub_mixin.py#L196

**Comment by julien-c** [comment]
**TL;DR:** Sounds good to me, do you want to give it a try?

---

Longer-term I think we should consider making `ModelHubMixin` more agnostic and **require** the user to define a `_save_pretrained`  and a `_from_pretrained`  method for it to work.

And then we would implement a `TorchModelHubMixin` , `ScikitLearnModelHubMixin` , etc. that are framework-specific and slightly more opinionated (and then in your case you can either tweak `TorchModelHubMixin`, or just inherit from the general ModelHubMixin and implement your logic the way you want)

**Comment by nateraw** [comment]
Sure, I'll give it a go!
